src/common/dao/webhook_job.go

```diff
@@ -0,0 +1,127 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+)
+
+// UpdateWebhookJob update webhook job
+func UpdateWebhookJob(job *models.WebhookJob, props ...string) (int64, error) {
+	if job.ID == 0 {
+		return 0, fmt.Errorf("webhook job ID is empty")
+	}
+
+	o := GetOrmer()
+	return o.Update(job, props...)
+}
+
+// UpdateWebhookJobStatus ...
+func UpdateWebhookJobStatus(id int64, status string, statusCondition ...string) (int64, error) {
+	qs := GetOrmer().QueryTable(&models.WebhookJob{}).Filter("id", id)
+	if len(statusCondition) > 0 {
+		qs = qs.Filter("status", statusCondition[0])
```

Comment:
If the `statusCondition` does not match nothing will be updated, is it the expected behavior?

Reply:
> If the `statusCondition` does not match nothing will be updated, is it the expected behavior?

yes